src/main/java/com/mraof/minestuck/item/ReadableSburbCodeItem.java

```diff
@@ -0,0 +1,161 @@
+package com.mraof.minestuck.item;
+
+import com.mraof.minestuck.client.gui.MSScreenFactories;
+import com.mraof.minestuck.util.BlockHitResultUtil;
+import com.mraof.minestuck.util.MSTags;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.core.Registry;
+import net.minecraft.nbt.*;
+import net.minecraft.nbt.ListTag;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TranslatableComponent;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResultHolder;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.TooltipFlag;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Block;
+import net.minecraftforge.registries.ForgeRegistries;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The class for the items Sburb Code and Completed Sburb Code which primarily allows them to be read, with Sburb Code extending its usage.
+ * Some checks for the use of sburb code exists in ComputerBlock, and functions from here are used in both ComputerBlock and ComputerTileEntity
+ */
+public class ReadableSburbCodeItem extends Item
+{
+	public ReadableSburbCodeItem(Item.Properties properties)
+	{
+		super(properties);
+	}
+	
+	@Override
+	public InteractionResultHolder<ItemStack> use(Level levelIn, Player playerIn, InteractionHand handIn)
```

Comment:
The alternative function `item.useOn()` seems to be designed for items right-clicked on blocks. Is there a reason that you're using `use()` instead of `useOn()`? If not, I'd suggest using that one instead.

Reply:
I dont remember if this was the reason originally, but I cant seem to get the screen to be openable when using useOn